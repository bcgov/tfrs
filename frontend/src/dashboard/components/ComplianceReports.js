import React from 'react'
import PropTypes from 'prop-types'

import Loading from '../../app/components/Loading'
import COMPLIANCE_REPORTING from '../../constants/routes/ComplianceReporting'
import { useNavigate } from 'react-router'

const ComplianceReports = (props) => {
  const { isFinding, supplementalItems, isGettingDashboard } = props.complianceReports
  const navigate = useNavigate()

  if (isFinding || isGettingDashboard) {
    return <Loading />
  }

  const awaitingReview = {
    analyst: 0,
    director: 0,
    manager: 0,
    total: 0
  }

  supplementalItems && supplementalItems.forEach((item) => {
    const { status } = item
    if (status.fuelSupplierStatus === 'Submitted' && status.analystStatus === 'Unreviewed' &&
        status.directorStatus === 'Unreviewed' && status.managerStatus === 'Unreviewed') {
      awaitingReview.analyst += 1
      awaitingReview.total += 1
    }

    if (['Not Recommended', 'Recommended'].indexOf(status.analystStatus) >= 0 &&
    status.managerStatus === 'Unreviewed' && status.directorStatus === 'Unreviewed' &&
    status.fuelSupplierStatus === 'Submitted') {
      awaitingReview.manager += 1
      awaitingReview.total += 1
    }

    if (['Not Recommended', 'Recommended'].indexOf(status.managerStatus) >= 0 &&
    status.directorStatus === 'Unreviewed') {
      awaitingReview.director += 1
      awaitingReview.total += 1
    }
  })

  return (
    <div className="dashboard-fieldset compliance-fieldset">
      <h1>Compliance &  Exclusion Reports</h1>
      There are:

      <div>
        <br/>
        <div className='compliance-fieldset-value'>
          <div className="value">
            {awaitingReview.total}
          </div>
        </div>
        <div className="content">
          <div>
            <button
              onClick={() => {
                props.setFilter([{
                  id: 'current-status',
                  value: ['For Analyst Review', 'For Manager Review', 'For Director Review']
                }
                ], 'compliance-reporting')

                return navigate(COMPLIANCE_REPORTING.LIST, { state: { items: ['For Analyst Review', 'For Manager Review', 'For Director Review'] } })
              }}
              type="button"
            >
             Compliance / Exclusion report(s) in progress:
            </button>
          </div>
        </div>
      </div>

      <div>
        <div className="content">
          <h2>View all reports:</h2>

          <div>
            <button
              onClick={() => {
                const currentYear = new Date().getFullYear()

                props.setFilter([{
                  id: 'compliance-period',
                  value: currentYear.toString()
                }], 'compliance-reporting')

                return navigate(COMPLIANCE_REPORTING.LIST)
              }}
              type="button"
            >
              Current compliance period
            </button>
            {' | '}
            <button
              onClick={() => {
                props.setFilter([{
                  id: 'compliance-period',
                  value: ''
                }], 'compliance-reporting')

                return navigate(COMPLIANCE_REPORTING.LIST)
              }}
              type="button"
            >
              All/historical
            </button>
          </div>
        </div>
      </div>
    </div>
  )
}

ComplianceReports.defaultProps = {
}

ComplianceReports.propTypes = {
  complianceReports: PropTypes.shape({
    isFetching: PropTypes.bool,
    isFinding: PropTypes.bool,
    isGettingDashboard: PropTypes.bool,
    items: PropTypes.arrayOf(PropTypes.shape()),
    supplementalItems: PropTypes.arrayOf(PropTypes.shape())
  }).isRequired,
  setFilter: PropTypes.func.isRequired
}

export default ComplianceReports
