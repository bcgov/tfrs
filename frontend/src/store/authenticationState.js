import { loadUser } from 'redux-oidc';
import { put, takeLatest, all } from 'redux-saga/effects';
import userManager from './oidc-usermanager';
import { getLoggedInUser } from '../actions/userActions';
import CONFIG from '../config';

const LOGIN_TRIGGERING_ACTIONS = [
  'redux-oidc/USER_EXPIRED'
];

function triggerLoginFlow (store) {
  const { routing } = store.getState();

  if (routing.location &&
    routing.location.pathname !== '/authCallback') {
    return CONFIG.KEYCLOAK.CUSTOM_LOGIN ? userManager.signinSilent() : userManager.signinRedirect();
  }

  return false;
}

function * getBackendUser () {
  yield put(getLoggedInUser());
}

export default function * authenticationStateSaga (store) {
  userManager.clearStaleState();

  const { routing } = store.getState();

  if (!routing.location || routing.location.pathname !== '/authCallback') {
    loadUser(store, userManager);
  }
  yield all([
    takeLatest('redux-oidc/USER_FOUND', getBackendUser),
    takeLatest(
      action => (LOGIN_TRIGGERING_ACTIONS.includes(action.type)),
      triggerLoginFlow,
      store
    )
  ]);
}
