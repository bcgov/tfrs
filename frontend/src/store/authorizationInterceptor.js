import axios from 'axios';
import userManager from './oidc-usermanager';
import CONFIG from '../config';

function configureAxios () {
  if (CONFIG.KEYCLOAK) {
    console.log('adding token to request');

    axios.interceptors.request.use((config) => {
      const loadconfig = async () => {
        const user = await userManager.getUser();
        return {
          ...config,
          headers: {
            Authorization: `Bearer ${user.id_token}`
          }
        };
      };

      return loadconfig();
    });
  }
}

export default configureAxios;
