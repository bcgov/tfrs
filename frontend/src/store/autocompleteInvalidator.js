import { call, put, takeLatest } from 'redux-saga/effects';
import AutocompleteActionTypes from "../constants/actionTypes/Autocomplete";
import FuelCodeActionTypes from '../constants/actionTypes/FuelCodes';

const INVALIDATING_ACTIONS = [
  FuelCodeActionTypes.ADD_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODES,
  FuelCodeActionTypes.UPDATE_FUEL_CODE,
  FuelCodeActionTypes.DELETE_FUEL_CODE
];

function * invalidateCache () {
  yield put({ type: AutocompleteActionTypes.INVALIDATE_AUTOCOMPLETE_CACHE });
}

export default function * sessionTimeoutSaga () {
  yield takeLatest(
    action => (INVALIDATING_ACTIONS.includes(action.type)),
    invalidateCache
  );
}
