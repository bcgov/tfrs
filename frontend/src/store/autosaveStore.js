import {takeEvery, all, call} from 'redux-saga/effects';
import {delay} from "redux-saga";
import axios from "axios";
import * as Routes from "../constants/routes";

function* loadAutosaveData(action) {
  const {key, resolve, reject} = action.payload;

  axios
    .get(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}?key=${key}`)
    .then((response) => {
      try {
        const parsed = JSON.parse(response.data.data);
        resolve(parsed);
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

function* saveAutosaveData(action) {
  const {key, state, resolve, reject} = action.payload;

  const payload = {
    key,
    data: JSON.stringify(state)
  };

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}`, payload)
    .then((response) => {
      resolve();
    }).catch((error) => {
    reject(error);
  });
}

function* clearAutosaveData(action) {
  const {resolve, reject} = action.payload;

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}/clear`)
    .then((response) => {
      try {
        resolve();
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

export default function* autosaveSaga() {

  yield all([
    takeEvery('LOAD_AUTOSAVE_DATA', loadAutosaveData),
    takeEvery('SAVE_AUTOSAVE_DATA', saveAutosaveData),
    takeEvery('CLEAR_AUTOSAVE_DATA', clearAutosaveData)
  ]);
}
