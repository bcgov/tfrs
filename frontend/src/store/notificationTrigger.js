import {call, put, takeLatest} from 'redux-saga/effects';
import {delay} from 'redux-saga';
import {getNotifications, getNotificationsCount} from "../actions/notificationActions";
import UserActionTypes from "../constants/actionTypes/Users";
import NotificationActionTypes from "../constants/actionTypes/Notifications";


const TRIGGERING_ACTIONS = [
  'SERVER_INITIATED_NOTIFICATION_RELOAD',
  UserActionTypes.RECEIVE_LOGGED_IN_USER,
  NotificationActionTypes.SUCCESS_NOTIFICATIONS
];

function* fetchNotifications(store) {
  yield call(delay, 1000); //debounce

  if (store.getState().rootReducer.userRequest.isAuthenticated) {
    yield put(getNotificationsCount());
  }

  if (store.getState().rootReducer.notifications.onNotificationsPage) {
    yield put(getNotifications());
  }

}

export default function* notificationsSaga(store) {
  yield takeLatest(
    action => (TRIGGERING_ACTIONS.includes(action.type)),
    fetchNotifications,
    store
  );
}
