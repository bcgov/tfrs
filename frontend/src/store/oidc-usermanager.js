import { createUserManager } from 'redux-oidc';
import CONFIG from '../config';

const settings = {
  authority: CONFIG.KEYCLOAK.AUTHORITY,
  client_id: CONFIG.KEYCLOAK.CLIENT_ID,
  redirect_uri: CONFIG.KEYCLOAK.CALLBACK_URL,
  automaticSilentRenew: true,
  silent_redirect_uri: `${window.location.protocol}//${window.location.hostname}${window.location.port ? `:${window.location.port}` : ''}/token_renew.html`,
  response_type: 'token id_token'
};

const userManager = createUserManager(settings);

export default userManager;
