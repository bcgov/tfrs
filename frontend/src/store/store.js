import { createStore, compose, applyMiddleware, combineReducers } from 'redux';
import { reducer as toastrReducer } from 'react-redux-toastr';
import { routerReducer, routerMiddleware } from 'react-router-redux';
import { createLogger } from 'redux-logger';
import io from 'socket.io-client';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/reducer';
import { getNotifications } from '../actions/notificationActions';
import { SOCKETIO_URL } from '../constants/routes';

import createOidcMiddleware, { loadUser, processSilentRenew, reducer as OIDCReducer } from 'redux-oidc';
import { persistTargetPathReducer } from '../reducers/persistTargetPathReducer';

import userManager from './oidc-usermanager';
import CONFIG from '../config';
import { getLoggedInUser } from '../actions/userActions';

import persistState from 'redux-localstorage';
import createSocketIoMiddleware from 'redux-socket.io';

const middleware = routerMiddleware(history);
const oidcMiddleware = createOidcMiddleware(userManager);

const socket = io(SOCKETIO_URL);
const socketIoMiddleware = createSocketIoMiddleware(socket, 'socketio/');

const enhancer = compose(persistState(['targetPath'], {key: 'tfrs-state'}));

const combinedReducers = (state = {}, action) => {
  const currentRoute = state.routing || {};
  return {
    toastr: toastrReducer(state.toastr, action),
    oidc: OIDCReducer(state.oidc, action),
    routing: routerReducer(state.routing, action),
    targetPath: persistTargetPathReducer(state.targetPath, {...action, currentRoute}),
    rootReducer: rootReducer(state.rootReducer, action),
  };
};

let allMiddleware = [
  thunk,
  socketIoMiddleware,
  oidcMiddleware,
  middleware
];

if (process.env.NODE_ENV !== 'production') {
  allMiddleware.push(createLogger());
}

const store = createStore(
  combinedReducers,
  applyMiddleware(...allMiddleware),
  enhancer
);

let subscriptionProcessing = false;

store.subscribe(() => {
  const state = store.getState();
  if (!subscriptionProcessing) {
    subscriptionProcessing = true;

    if (state.rootReducer.notifications.serverInitiatedReloadRequested === true) {
      store.dispatch(getNotifications());
    }

    if (CONFIG.KEYCLOAK.ENABLED) {
      if (state.oidc.user &&
        !state.rootReducer.userRequest.isFetching &&
        !state.rootReducer.userRequest.isAuthenticated) {
        store.dispatch(getLoggedInUser());
      }
    }

    subscriptionProcessing = false;
  }
});

if (CONFIG.KEYCLOAK.ENABLED) {
  loadUser(store, userManager);
  processSilentRenew();
}

export default store;
