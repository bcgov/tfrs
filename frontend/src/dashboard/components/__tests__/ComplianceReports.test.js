import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { screen, render, fireEvent } from '@testing-library/react'
import store from '../../../store/store'
import ComplianceReports from '../ComplianceReports'

describe('ComplianceReports', () => {
  test('Should render the component', () => {
    const props = {
      complianceReports: {
        isFinding: false,
        isGettingDashboard: false,
        items: []
      },
      setFilter: jest.fn()
    }
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ComplianceReports {...props} />
        </Provider>
      </BrowserRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  //   test('Checking the Analyst Count', () => {
  //     const props = {
  //       complianceReports: {
  //         isFinding: false,
  //         isGettingDashboard: false,
  //         items: [
  //           {
  //             id: 654,
  //             status: {
  //               analystStatus: 'Unreviewed',
  //               directorStatus: 'Unreviewed',
  //               fuelSupplierStatus: 'Submitted',
  //               managerStatus: 'Unreviewed',
  //             },
  //             type: 'Compliance Report',
  //             supplementalReports: [],
  //           },
  //         ],
  //       },
  //       setFilter: jest.fn(),
  //     }
  //     const component = renderer.create(
  //       <BrowserRouter>
  //         <Provider store={store}>
  //           <ComplianceReports {...props} />
  //         </Provider>
  //       </BrowserRouter>
  //     )
  //     const tree = component.toJSON()
  //     expect(tree).toMatchSnapshot()
  //   })

  //   test('Checking the Manager Count', () => {
  //     const props = {
  //       complianceReports: {
  //         isFinding: false,
  //         isGettingDashboard: false,
  //         items: [
  //           {
  //             id: 294,
  //             status: {
  //               analystStatus: 'Unreviewed',
  //               directorStatus: 'Unreviewed',
  //               managerStatus: 'Unreviewed',
  //             },
  //             type: 'Compliance Report',
  //             supplementalReports: [
  //               {
  //                 status: {
  //                   analystStatus: 'Recommended',
  //                   directorStatus: 'Unreviewed',
  //                   managerStatus: 'Unreviewed',
  //                 },
  //               },
  //             ],
  //           },
  //           {
  //             id: 295,
  //             status: {},
  //             type: 'Exclusion Report',
  //             supplementalReports: [
  //               {
  //                 status: {},
  //                 supplementalReports: [
  //                   {
  //                     status: {
  //                       analystStatus: 'Recommended',
  //                       directorStatus: 'Unreviewed',
  //                       managerStatus: 'Unreviewed',
  //                     },
  //                   },
  //                 ],
  //               },
  //             ],
  //           },
  //         ],
  //       },
  //       setFilter: jest.fn(),
  //     }
  //     const component = renderer.create(
  //       <BrowserRouter>
  //         <Provider store={store}>
  //           <ComplianceReports {...props} />
  //         </Provider>
  //       </BrowserRouter>
  //     )
  //     const tree = component.toJSON()
  //     expect(tree).toMatchSnapshot()
  //   })

  //   test('Checking the Director Count', () => {
  //     const props = {
  //       complianceReports: {
  //         isFinding: false,
  //         isGettingDashboard: false,
  //         items: [
  //           {
  //             id: 294,
  //             status: {
  //               analystStatus: 'Unreviewed',
  //               directorStatus: 'Unreviewed',
  //               managerStatus: 'Unreviewed',
  //             },
  //             type: 'Compliance Report',
  //             supplementalReports: [
  //               {
  //                 status: {
  //                   analystStatus: 'Recommended'',
  //                   directorStatus: 'Unreviewed',
  //                   managerStatus: 'Not Recommended',
  //                 },
  //               },
  //             ],
  //           },
  //         ],
  //       },
  //       setFilter: jest.fn(),
  //     }
  //     const component = renderer.create(
  //       <BrowserRouter>
  //         <Provider store={store}>
  //           <ComplianceReports {...props} />
  //         </Provider>
  //       </BrowserRouter>
  //     )
  //     const tree = component.toJSON()
  //     expect(tree).toMatchSnapshot()
  //   })

  test('Checking the analyst count click event', () => {
    const props = {
      complianceReports: {
        isFinding: false,
        isGettingDashboard: false,
        items: [
          {
            id: 654,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              fuelSupplierStatus: 'Submitted',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: []
          },
          {
            id: 655,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              fuelSupplierStatus: 'Submitted',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: []
          },
          {
            id: 656,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              fuelSupplierStatus: 'Submitted',
              managerStatus: 'Unreviewed'
            },
            type: 'Exclusion Report',
            supplementalReports: []
          }
        ]
      },
      setFilter: jest.fn()
    }
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ComplianceReports {...props} />
        </Provider>
      </BrowserRouter>
    )
    const complianceBtnEl = screen.getByRole('button', {
      name: '2 awaiting government analyst review'
    })
    const exclusionBtnEl = screen.getByRole('button', {
      name: '1 awaiting government analyst review'
    })
    const complianceBtn = fireEvent.click(complianceBtnEl)
    const exclusionBtn = fireEvent.click(exclusionBtnEl)
    expect(complianceBtn).toBe(true)
    expect(exclusionBtn).toBe(true)
    expect(props.setFilter).toHaveBeenCalledTimes(2)
  })

  test('Checking the Manager Count click', () => {
    const props = {
      complianceReports: {
        isFinding: false,
        isGettingDashboard: false,
        items: [
          {
            id: 293,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: [
              {
                status: {
                  analystStatus: 'Recommended',
                  directorStatus: 'Unreviewed',
                  managerStatus: 'Unreviewed'
                }
              }
            ]
          },
          {
            id: 294,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: [
              {
                status: {
                  analystStatus: 'Recommended',
                  directorStatus: 'Unreviewed',
                  managerStatus: 'Unreviewed'
                }
              }
            ]
          },
          {
            id: 295,
            status: {},
            type: 'Exclusion Report',
            supplementalReports: [
              {
                status: {},
                supplementalReports: [
                  {
                    status: {
                      analystStatus: 'Recommended',
                      directorStatus: 'Unreviewed',
                      managerStatus: 'Unreviewed'
                    }
                  }
                ]
              }
            ]
          }
        ]
      },
      setFilter: jest.fn()
    }
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ComplianceReports {...props} />
        </Provider>
      </BrowserRouter>
    )
    const complianceBtnEl = screen.getByRole('button', {
      name: '2 awaiting compliance manager review'
    })
    const exclusionBtnEl = screen.getByRole('button', {
      name: '1 awaiting compliance manager review'
    })
    const complianceBtn = fireEvent.click(complianceBtnEl)
    const exclusionBtn = fireEvent.click(exclusionBtnEl)
    expect(complianceBtn).toBe(true)
    expect(exclusionBtn).toBe(true)
    expect(props.setFilter).toHaveBeenCalledTimes(2)
  })

  test('Checking the Director Count click', () => {
    const props = {
      complianceReports: {
        isFinding: false,
        isGettingDashboard: false,
        items: [
          {
            id: 294,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: [
              {
                status: {
                  analystStatus: 'Recommended',
                  directorStatus: 'Unreviewed',
                  managerStatus: 'Not Recommended'
                }
              }
            ]
          },
          {
            id: 295,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              managerStatus: 'Unreviewed'
            },
            type: 'Compliance Report',
            supplementalReports: [
              {
                status: {
                  analystStatus: 'Recommended',
                  directorStatus: 'Unreviewed',
                  managerStatus: 'Not Recommended'
                }
              }
            ]
          },
          {
            id: 296,
            status: {
              analystStatus: 'Unreviewed',
              directorStatus: 'Unreviewed',
              managerStatus: 'Unreviewed'
            },
            type: 'Exclusion Report',
            supplementalReports: [
              {
                status: {
                  analystStatus: 'Recommended',
                  directorStatus: 'Unreviewed',
                  managerStatus: 'Not Recommended'
                }
              }
            ]
          }
        ]
      },
      setFilter: jest.fn()
    }
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ComplianceReports {...props} />
        </Provider>
      </BrowserRouter>
    )
    const complianceBtnEl = screen.getByRole('button', {
      name: '2 awaiting Director review'
    })
    const exclusionBtnEl = screen.getByRole('button', {
      name: '1 awaiting Director review'
    })
    const complianceBtn = fireEvent.click(complianceBtnEl)
    const exclusionBtn = fireEvent.click(exclusionBtnEl)
    expect(complianceBtn).toBe(true)
    expect(exclusionBtn).toBe(true)
    expect(props.setFilter).toHaveBeenCalledTimes(2)
  })

  test('Check Current Compliance Period Btn and All/historical Btn', () => {
    const props = {
      complianceReports: {
        isFinding: false,
        isGettingDashboard: false,
        items: []
      },
      setFilter: jest.fn()
    }
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ComplianceReports {...props} />
        </Provider>
      </BrowserRouter>
    )
    const complianceBtnEl = screen.getByText('Current compliance period')
    const historicalBtnEl = screen.getByText('All/historical')
    const complianceBtn = fireEvent.click(complianceBtnEl)
    expect(complianceBtn).toBe(true)
    const historicalBtn = fireEvent.click(historicalBtnEl)
    expect(historicalBtn).toBe(true)
    expect(props.setFilter).toHaveBeenCalledTimes(2)
  })
})
