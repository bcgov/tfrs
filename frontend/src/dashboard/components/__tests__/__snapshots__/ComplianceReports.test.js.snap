// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplianceReports Should render the component 1`] = `
<div
  className="dashboard-fieldset"
>
  <h1>
    Compliance &amp Exclusion Reports
  </h1>
  There are:
  <div>
    <div
      className="value"
    >
      0
    </div>
    <div
      className="content"
    >
      <h2>
        compliance reports in progress:
      </h2>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting government analyst review
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting compliance manager review
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting Director review
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="value"
    >
      0
    </div>
    <div
      className="content"
    >
      <h2>
        exclusion reports in progress:
      </h2>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting government analyst review
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting compliance manager review
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          0
           awaiting Director review
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="content"
    >
      <h2>
        View all reports:
      </h2>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          Current compliance period
        </button>
         | 
        <button
          onClick={[Function]}
          type="button"
        >
          All/historical
        </button>
      </div>
    </div>
  </div>
</div>
`;
